Mergin/plugin.py

```diff
@@ -575,71 +625,116 @@ def __init__(self, parent, grp_name, grp_filter, icon, order, project_manager):
         self.setSortKey(order)
         self.project_manager = project_manager
-
-    def createChildren(self):
-        mc = self.parent().mc
-        if not mc:
+        self.projects = []
+        self.group_name = grp_name
+        self.total_projects_count = None
+        self.fetch_more_item = None
+
+    def fetch_projects(self, page=1, per_page=PROJS_PER_PAGE):
+        """Get paginated projects list from Mergin service and an error message list."""
+        if self.project_manager is None:
             error_item = QgsErrorItem(self, "Failed to login please check the configuration", "/Mergin/error")
             sip.transferto(error_item, self)
             return [error_item]
         try:
-            projects = mc.projects_list(flag=self.filter)
+            resp = self.project_manager.mc.paginated_projects_list(
+                flag=self.filter, page=page, per_page=per_page, order_by="namespace")
+            self.projects += resp["projects"]
+            self.total_projects_count = int(resp["count"]) if is_number(resp["count"]) else 0
         except URLError:
             error_item = QgsErrorItem(self, "Failed to get projects from server", "/Mergin/error")
             sip.transferto(error_item, self)
             return [error_item]
         except Exception as err:
             error_item = QgsErrorItem(self, "Error: {}".format(str(err)), "/Mergin/error")
             sip.transferto(error_item, self)
-            return [error_item]
+            return None, [error_item]
+        return None
 
+    def createChildren(self):
+        if not self.projects:
+            error = self.fetch_projects()
+            if error:
+                return error
         items = []
-        for project in projects:
-            item = MerginProjectItem(self, project, mc, self.project_manager)
+        for project in self.projects:
+            item = MerginProjectItem(self, project, self.project_manager)
             item.setState(QgsDataItem.Populated)  # make it non-expandable
             sip.transferto(item, self)
             items.append(item)
+        self.set_fetch_more_item()
+        if self.fetch_more_item is not None:
+            items.append(self.fetch_more_item)
         return items
 
+    def set_fetch_more_item(self):
+        """Check if there are more projects to be fetched from Mergin service and set the fetch-more item."""
+        if self.fetch_more_item is not None:
+            try:
+                self.removeChildItem(self.fetch_more_item)
+            except RuntimeError:
+                pass
+            self.fetch_more_item = None
+        fetched_count = self.rowCount() + len(self.projects)
+        if fetched_count < self.total_projects_count:
+            self.fetch_more_item = FetchMoreItem(self)
+            self.fetch_more_item.setState(QgsDataItem.Populated)
+            sip.transferto(self.fetch_more_item, self)
+        group_name = f"{self.group_name} ({fetched_count}/{self.total_projects_count})"
```

Comment:
I don't really like the idea to show "XXX (fetched/total)" for each group, sorry :-)  maybe just show "XXX (total)" but including the number of fetched projects is IMHO not needed...

Reply:
I was think about users fetching more of Explore group projects, so they clearly know how many of them are already fetched - there are nearly 1K of them ATM. It is true that users have the projects ordered alphabetically and can kind of guess that there is plenty of them to be fetched if they see D as the first letter of projects name for example.
It is also true, that there won't be Fetch more... item at the bottom of the list if there is nothing more to get.
And also true that for "My Projects" and "Shared with me" projects there not that many of them and it looks a bit odd.

I will show only the total number of projects in each group, as you suggest.
How about having the complete count info in the tool tip for group items?